src/test/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheckTest.java

```diff
@@ -180,6 +180,18 @@ public void testFileWithEmptyLineOnly() throws Exception {
                 expected);
     }
 
+    @Test
+    public void testFileWithEmptyLineOnlyWithLfCrCrlf() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NewlineAtEndOfFileCheck.class);
+        checkConfig.addAttribute("lineSeparator", LineSeparatorOption.LF_CR_CRLF.toString());
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verify(
+                createChecker(checkConfig),
+                getPath("InputNewlineAtEndOfFileNewlineAtEnd.txt"),
```

Comment:
> /var/tmp $ xxd foo.properties
00000000: 0a       

The test case between the issue and here is different. `InputNewlineAtEndOfFileNewlineAtEnd` will have `\n` or `\r\n` depending on the OS.

Does the issue still happen with this file even though the file is longer than 1 byte?
Do we want to use a file that is the same line encoding on all OSes that is controlled through the `.gitattributes` file like `InputNewlineAtEndOfFileLf`?